client/web/compose/src/components/PageBlocks/Navigation/Base.vue

```diff
@@ -0,0 +1,136 @@
+          :target="list.options.targetType"
+        >
+          <template v-if="list.navigationType === 'text-type'">
+            <span>
+              {{ list.options.itemOption.label }}
+            </span>
+          </template>
+          <template v-if="list.navigationType === 'url-type'">
+            <span>
+              {{ list.options.itemOption.label }}
+            </span>
+          </template>
+          <template v-if="(list.navigationType === 'compose-type' && !list.options.itemOption.displaySubPages)">
+            <span>
+              {{ list.options.itemOption.label }}
+            </span>
+          </template>
+        </b-nav-item>
+
+        <b-nav-item-dropdown
+          v-for="(list, index) in options.lists"
+          v-show="list.navigationType === 'dropdown-type'"
+          :key="`lists-dropdown-${index}`"
+          :text="list.options.dropdownText"
+          :variant="list.options.bgVariant"
+          :right="list.options.dropdownAlignType === 'right'"
+          :left="list.options.dropdownAlignType === 'left'"
+          :style="{ order: index }"
+          class="nav-item-dropdown"
+        >
+          <b-dropdown-item
+            v-for="(dropdown, dIndex) in list.options.dropdown"
+            :key="`dropdown-${dIndex}`"
+            :to="dropdown.url"
+            :disabled="list.options.disabled"
+            :style="{ order: dIndex * 2 }"
+          >
+            {{ dropdown.label }}
+          </b-dropdown-item>
+
+          <b-dropdown-divider
+            v-for="(dropdown, dIndex) in list.options.dropdown"
+            v-show="dropdown.delimiter"
+            :key="`dropdown-divider-${dIndex}`"
+            :style="{ order: dIndex + 1 }"
+          />
+        </b-nav-item-dropdown>
+
+        <b-nav-item-dropdown
+          v-for="(list, index) in options.lists"
+          v-show="(list.navigationType === 'compose-type' && list.options.itemOption.displaySubPages)"
+          :key="`lists-dropdown-compose-${index}`"
+          :text="list.options.itemOption.label"
+          :variant="list.options.bgVariant"
+          :style="{ order: index }"
+          class="nav-item-dropdown"
+        >
+          <b-dropdown-item
+            v-for="(dropdown, dIndex) in getSubPages(list.options.itemOption.referenceId)"
+            :key="`dropdown-${dIndex}`"
+            :disabled="list.options.disabled"
+            :to="dropdown.pageID"
+          >
+            {{ dropdown.title }}
+          </b-dropdown-item>
+        </b-nav-item-dropdown>
+      </b-nav>
+    </div>
+  </wrap>
+</template>
+<script>
+import base from '../base'
+
+export default {
+  extends: base,
+
+  methods: {
+    processLink (list) {
+      if (list.navigationType === 'compose') {
+        return list.options.itemOption.referenceId
+      }
+
+      return ''
+    },
+
+    processHrefLink (list) {
+      if (list.navigationType === 'url') {
+        return list.options.itemOption.url
```

Comment:
wouldn't the item.url be undefined anyway so you can just do return item.url.
You can add || '' if you need it to be a string